dlt/common/libs/pyarrow.py

```diff
@@ -1099,3 +1099,61 @@ def add_arrow_metadata(
         return pyarrow.Table.from_arrays(item.columns, schema=new_schema)
     else:  # RecordBatch
         return pyarrow.RecordBatch.from_arrays(item.columns, schema=new_schema)
+
+
+def _has_offset_timezones(item: Union[pyarrow.Table, pyarrow.RecordBatch]) -> bool:
+    """Eager check to identify it the table contains fields with offset-based timezones."""
+    for field in item.schema:
+        if getattr(field.type, "tz", "").startswith(("+", "-")):
+            return True
+    return False
+
+
+# NOTE could use an explicit mapping of ~30 offset to IANA timezones
+# would be less error-prone and easy to maintain.
+def _offset_to_iana(offset_tzinfo: str) -> str:
+    """Convert an offset-based time zone to IANA time zone.
+
+    This function assumes that the offset time zone is valid.
+
+    Facts:
+    - Time zones go from `-12:00` to `+14:00`
+    - Neutral time zone is `+00:00`
+    - The sign used in UTC is reversed for GMT; for example: UTC+05:00 == Etc/GMT-5
+    ref: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+    """
+    hours, _, minutes = offset_tzinfo[1:].partition(":")
+    sign = "-" if offset_tzinfo[0] == "+" and hours != "00" else "+"
+    if int(minutes) != 0:
```

Comment:
really? so IST (India) is not supported? `UTC +05:30` this is super common I suppose, most of people live in India ). Nepal has UTC +5:45. is this a limitation of the c library underneath Arrow?